code/modules/mob/living/simple_animal/hostile/megafauna/_megafauna.dm

```diff
@@ -169,11 +169,13 @@
 
 /mob/living/simple_animal/hostile/megafauna/CanAttack(atom/the_target)
 	. = ..()
+	var/mob/living/living_target = the_target
+	if(living_target.has_status_effect(/datum/status_effect/shapechange_mob/from_spell && !isnull(living_target.ckey))) // Prevents polymorph belt cheese
+		return TRUE
```

Comment:
This should be down where your deleted line is, not before the early returns.

Reply:
also this is why I hate simple mobs